pkg/datapath/loader/netlink.go

```diff
@@ -98,6 +98,33 @@ func replaceDatapath(ctx context.Context, ifName, objPath string, progs []progDe
 		}
 	}
 
+	// Unconditionally repin cilium_calls_* maps to prevent them from being
+	// repopulated by the loader.
+	for key, ms := range spec.Maps {
+		if !strings.HasPrefix(ms.Name, "cilium_calls_") {
+			continue
+		}
+
+		if err := bpf.RepinMap(bpf.TCGlobalsPath(), key, ms); err != nil {
+			return nil, fmt.Errorf("repinning map %s: %w", key, err)
+		}
+
+		defer func() {
+			revert := false
+			// This captures named return variable err.
+			if err != nil {
+				revert = true
```

Comment:
I'm not entirely sure if that is possible but what happens if we need to revert the unconditional migration on a cilium_calls_* map but the map has been recreated and pinned through `bpf.LoadCollection` already? AFAIU `bpf.FinalizeMap()` will fail and we log an error but is there any danger when this happens?

Reply:
That's the change I wanted to make while you were reviewing today; `FinalizeMap()` now does an `os.Remove()` in the revert path.